mailbox/api/src/test/java/org/apache/james/mailbox/events/EventBusContract.java

```diff
@@ -80,6 +86,132 @@ default MailboxListener newListener() {
+            EventCollector eventCollector = eventCollector();
+
+            doThrow(new RuntimeException())
+                .doThrow(new RuntimeException())
+                .doThrow(new RuntimeException())
+                .doCallRealMethod()
+                .when(eventCollector).event(EVENT);
+
+            eventBus().register(eventCollector, new GroupA());
+            eventBus().dispatch(EVENT, NO_KEYS).block();
+
+            assertThat(eventCollector.getEvents())
+                .hasSize(1);
+        }
+
+        @Test
+        default void listenerShouldNotReceiveWhenFailsGreaterThanMaxRetries() {
+            EventCollector eventCollector = eventCollector();
+
+            doThrow(new RuntimeException())
+                .doThrow(new RuntimeException())
+                .doThrow(new RuntimeException())
+                .doThrow(new RuntimeException())
+                .doCallRealMethod()
+                .when(eventCollector).event(EVENT);
+
+            eventBus().register(eventCollector, new GroupA());
+            eventBus().dispatch(EVENT, NO_KEYS).block();
+
+            assertThat(eventCollector.getEvents())
+                .isEmpty();
+        }
+
+        @Test
+        default void dispatchShouldTakeTimeToDoRetries() {
+            ThrowingListener throwingListener = throwingListener();
+
+            eventBus().register(throwingListener, new GroupA());
+            long dispatchingInMs = recordTimeRun(() -> eventBus().dispatch(EVENT, NO_KEYS).block());
+
+            SoftAssertions.assertSoftly(softly -> {
+                softly.assertThat(throwingListener.timeElapsed).hasSize(4);
+                long firstRetryHappenAfter = throwingListener.timeElapsed.get(1) - throwingListener.timeElapsed.get(0);
+                long secondRetryHappenAfter = throwingListener.timeElapsed.get(2) - throwingListener.timeElapsed.get(1);
+                long thirdRetryHappenAfter = throwingListener.timeElapsed.get(3) - throwingListener.timeElapsed.get(2);
+
+                // some CPU computation capabilities could not bring exactly delay expected but bit longer, relax jitter in assertions
+                double acceptedJitterFactor = DEFAULT_JITTER_FACTOR + 0.1;
+                long minFirstDelay = 100;
+                long minSecondDelay = 200 - Double.valueOf(200 * acceptedJitterFactor).longValue();
+                long minThirdDelay = 400 - Double.valueOf(400 * acceptedJitterFactor).longValue();
+                softly.assertThat(dispatchingInMs)
+                    .isGreaterThanOrEqualTo(minFirstDelay + minSecondDelay + minThirdDelay);
+
+                Percentage acceptedDiffPercentage = Percentage.withPercentage(100 * acceptedJitterFactor);
+                softly.assertThat(firstRetryHappenAfter)
+                    .isGreaterThanOrEqualTo(100)
+                    .isCloseTo(100, acceptedDiffPercentage); // 100 <= firstRetryHappenAfter <= 150 by theory
```

Comment:
IMO you sould assert:

```
// first retry happends at t1 > t0 + 100ms
// second retry happened at t2 > t1 + 200ms
// third retry at t3> t2 + 400 ms
```

We don't care that is is close to, and we don't want our test validity to be jeoparadized by poor performance / sketchy scheduling.

Let's reduce what we assert for more safety.

Reply:
I picked test assertions from reactor repository https://github.com/reactor/reactor-core/blob/20e155eeff3758bd6dfbb602e0237a2c28f5d267/reactor-core/src/test/java/reactor/core/publisher/MonoRetryWhenTest.java#L127

Reply:
Okay my remark is: do not check 'the retry happened before that time'

Always assert 'the retry happened after x time' because this is what we care about